From 1545bc931d037a17d5cc424774c5ed1a2a3f580f Mon Sep 17 00:00:00 2001
From: PrimeDirective <activethrasher00@gmail.com>
Date: Mon, 30 Sep 2013 13:41:16 -0400
Subject: [PATCH 1/6] Scroll: modified velocitytracker

Velocitytracker and overscroll are called on every touch event to
calculate if there is a scrolling/fling. Move this and the recycling
to static voids so they are pre-optimized at compile time rather than
after evoking the method call during runtime (JIT) to increase responsiveness.

-Vanir

Change-Id: Ic7351da58f29281ca2b6a788386b95b68b552390

Conflicts:
	core/java/android/widget/ScrollView.java

Signed-off-by: mydongistiny <jaysonedson@gmail.com>
---
 core/java/android/widget/ScrollView.java | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/core/java/android/widget/ScrollView.java b/core/java/android/widget/ScrollView.java
index 97d32f1266a..61d3dae8ff7 100644
--- a/core/java/android/widget/ScrollView.java
+++ b/core/java/android/widget/ScrollView.java
@@ -84,7 +84,7 @@ public class ScrollView extends FrameLayout {
     private long mLastScroll;
 
     private final Rect mTempRect = new Rect();
-    private OverScroller mScroller;
+    private static OverScroller mScroller;
     private EdgeEffect mEdgeGlowTop;
     private EdgeEffect mEdgeGlowBottom;
 
@@ -116,7 +116,7 @@ public class ScrollView extends FrameLayout {
     /**
      * Determines speed during touch scrolling
      */
-    private VelocityTracker mVelocityTracker;
+    private static VelocityTracker mVelocityTracker;
 
     /**
      * When set to true, the scroll view measure its child to make it fill the currently
@@ -189,7 +189,6 @@ public class ScrollView extends FrameLayout {
                 attrs, com.android.internal.R.styleable.ScrollView, defStyleAttr, defStyleRes);
 
         setFillViewport(a.getBoolean(R.styleable.ScrollView_fillViewport, false));
-
         a.recycle();
 
         if (context.getResources().getConfiguration().uiMode == Configuration.UI_MODE_TYPE_WATCH) {
@@ -465,7 +464,7 @@ public class ScrollView extends FrameLayout {
         }
     }
 
-    private void recycleVelocityTracker() {
+    private static void recycleVelocityTracker() {
         if (mVelocityTracker != null) {
             mVelocityTracker.recycle();
             mVelocityTracker = null;
-- 
2.11.0

